core/sr-io/src/lib.rs

```diff
@@ -245,32 +275,28 @@ export_api! {
 		/// Returns information about the local node's network state.
 		fn network_state() -> Result<OpaqueNetworkState, ()>;
 
-		/// Returns the currently configured authority public key, if available.
-		fn pubkey(key: CryptoKey) -> Result<Vec<u8>, ()>;
-
-		/// Create new key(pair) for signing/encryption/decryption.
+		/// Create new key pair for signing/encryption/decryption.
 		///
 		/// Returns an error if given crypto kind is not supported.
-		fn new_crypto_key(crypto: CryptoKind) -> Result<CryptoKey, ()>;
+		fn new_key(crypto: CryptoKind, app_id: KeyTypeId) -> Result<CryptoKey, ()>;
```

Comment:
why is this still here?

now that we have `sr25519_generate` and `ed25519_generate`, it shouldn't be needed, right?

Reply:
Yes, the offchain stuff still needs a little bit love by me.